src/go/k8s/apis/redpanda/v1alpha1/cluster_webhook.go

```diff
@@ -855,6 +872,76 @@ func validateListener(
 				external.Subdomain,
 				"TLS requires specifying a subdomain"))
 	}
+	if tlsEnabled && clientCACertRef != nil {
```

Comment:
What if we simplify this by exiting early?

```suggestion
	if !tlsEnabled || clientCACertRef == nil {
		return allErrs
	}
```

Reply:
Agree.

Reply:
Can you address this comment?

Reply:
It has been done. 
See https://github.com/paulzhang97/redpanda/blob/9c814e4a6bd3d6321df8a01fe9f3961c73fefaa6/src/go/k8s/apis/redpanda/v1alpha1/cluster_webhook.go#L885-L887